db/db_impl/db_impl.cc

```diff
@@ -526,25 +528,28 @@ Status DBImpl::CloseHelper() {
     mutex_.Lock();
   }
 
-  for (auto l : logs_to_free_) {
-    delete l;
-  }
-  for (auto& log : logs_) {
-    uint64_t log_number = log.writer->get_log_number();
-    Status s = log.ClearWriter();
-    if (!s.ok()) {
-      ROCKS_LOG_WARN(
-          immutable_db_options_.info_log,
-          "Unable to Sync WAL file %s with error -- %s",
-          LogFileName(immutable_db_options_.wal_dir, log_number).c_str(),
-          s.ToString().c_str());
-      // Retain the first error
-      if (ret.ok()) {
-        ret = s;
+  {
+    InstrumentedMutexLock lock(&log_write_mutex_);
+    for (auto l : logs_to_free_) {
```

Comment:
What's the benefit to make logs_to_free_ be protected by both  log_write_mutex_ and mutex_

Reply:
It makes no sense. But i think unlock `mutex_` during this function may cause other bad case.